docs(aws): dual-region setup in AWS

Langleu opened the pull request:
## Description

related to: https://github.com/camunda/team-infrastructure-experience/issues/98

This is part of a series of upcoming pull requests concerning the multi-region, particularly dual-region, topic.
I'd target everything at a dedicated branch and later merge everything as a single PR to be rolled out later. The soft target is the 8.5 release.

This PR contains the AWS dual-region setup with Terraform on how to get it working.

```
npm run start
```

[link to page](http://localhost:3000/docs/next/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region)

Based on later reviews by QA / Support, I would add a troubleshooting section.

<!-- This helps the reviewers by providing an overview of what to expect in the PR. Linking to an issue is preferred but not required. -->
<!-- Add an assignee and use component: labels for good hygiene! -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->
<!-- Help the DevEx team prioritize our work (reviews, merges, etc.) by opening PRs sooner. -->

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

<!-- Keep in mind, Camunda maintains 18 months of versions. Backporting your change or including it in multiple versions is common. -->

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+export CAMUNDA_NAMESPACE_2=camunda-paris
+```
+
+## Installing Amazon EKS clusters with Terraform
````
## **[big one]**

I have a general feeling it's too much work for us to guide the user what files to create and what content to put there.

Maybe we can say the user needs to clone https://github.com/camunda/c8-multi-region/tree/aws-operational/test/resources/aws/2-region/terraform (with all files pre-created) and customize the Terraform variables and then just `tf apply`?

It's cool we can embed the code content, but I also have a feeling it won't be much useful for the user and create a lot of work for us to maintain/create such content.

wdyt?

Langleu replied:
the proposed change is included in the latest commit - https://github.com/camunda/camunda-docs/pull/3430/commits/4e11aa0a397cdf49f8810c07b254f8420fb23e58

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+The address forms as follows:
+
+```bash
+$pod.$service.$namespace.svc.cluster.local:26502
````
can something like https://github.com/camunda-community-hub/camunda-8-helm-profiles/blob/main/google/multi-region/active-active/setup-zeebe.py be used as a endpoint hostnames list generator?

i.e. a script put in https://github.com/camunda/c8-multi-region/tree/aws-operational/test/resources/aws/2-region/terraform and asking some basic inputs from a user / env vars (namespace name, broker count ...). It's a bit hard to generate those by hand without mistakes.

Langleu replied:
yes, we can probably have a small python script.

maxdanilov replied:
a Python script to run would be yet another thing easier with the `git clone` approach should we go for it.

Langleu replied:
was handled in - https://github.com/camunda/c8-multi-region/pull/17

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+</TabItem>
+<TabItem value="helm-chart" label="Camunda 8 Helm Chart prerequisites">
+
+Our recommendation is to work with layered helm values files since you will want to have a base `value.yml` that is generally applicable and then two overlays that are for `region0` and `region1`.
```
## [big one]

Same suggestion as for the TF files - does it make sense to instead to the users to `git clone` https://github.com/camunda/c8-multi-region/tree/aws-operational/test/resources/aws/2-region/kubernetes, adjust the scripts accordingly, so we don't need to explain what files with what contents to create?

Langleu replied:
the proposed change is included in the latest commit - https://github.com/camunda/camunda-docs/pull/3430/commits/4e11aa0a397cdf49f8810c07b254f8420fb23e58

christinaausley commented on optimize_sidebars.js:
```diff
@@ -1841,6 +1841,10 @@ module.exports = {
                       "self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/eks-helm/"
                     ),
+                    docsLink(
+                      "Deploy two EKS clusters with Terraform for a dual-region setup",
```
Will need to adjust title here as appropriate.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+https://github.com/camunda/c8-multi-region/blob/main/test/resources/aws/2-region/terraform/config.tf#L5-L51
+```
+
+  </summary>
````
will keep it open for now, not sure whether still needed with the current suggestions.
Never noticed since I always open links with the scroll wheel press.

maxdanilov replied:
Agree. Can be a bug of the Docusaurus plugging where they did not think of the corner case of when a code block is inside a collapsing block.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+There are several ways to authenticate the `AWS` provider.
+
+- (Recommended) Use the [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html) to configure access. Terraform will automatically default to AWS CLI configuration when present.
+- Set environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`, where the `key` and `id` can be retrieved from the [AWS Console](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html).
```
adjusted it also in the 8.4 docs for the terraform guide.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+
+:::warning
+
+Do not use secrets in your configuration files.
```
adjusted it for the 8.4 terraform guide as well.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+
+:::warning
+
+The user who creates the resources will always be the owner. This means the user will always have admin access to the Kubernetes cluster until you delete it. Therefore, it can make sense to create an extra [AWS IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html) that's solely used for Terraform purposes.
```
adjusted it for the 8.4 terraform guide as well.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+
+The user who creates the resources will always be the owner. This means the user will always have admin access to the Kubernetes cluster until you delete it. Therefore, it can make sense to create an extra [AWS IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html) that's solely used for Terraform purposes.
+
+:::
```
thought about that as well already to have a `next step` link switching the tab.
Yes Deeplinks work

Langleu replied:
keeping open for now, have to see how I approach this.
The querystring stuff will require a reload of the page and does not jump to the position.

The alternative is to simulate a click with javascript when clicking on a link.

Langleu replied:
tabs removed

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+  <summary>
+
+```hcl reference title="cluster.tf"
+https://github.com/camunda/c8-multi-region/blob/main/test/resources/aws/2-region/terraform/config.tf#L57-L86
````
the version is a linting thing, we can disable it.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+camunda-paris.svc.cluster.local:53 {
+  errors
+  cache 30
+  forward . 10.202.30.193 10.202.46.72 10.202.65.75 {
```
added an admonitions above it to cover the following 3 comments.

tl;dr
- Just example values
- Make sure to edit
- Won't work in your setup

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+
+kubectl --context $CLUSTER_2 exec -n $CAMUNDA_NAMESPACE_2 -it sample-nginx -- curl http://sample-nginx.sample-nginx-peer.camunda-london.svc.cluster.local
+
+# http://sample-nginx.sample-nginx-peer.camunda-london.svc.cluster.local
```
not needed, was just to showcase how it's constructed.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+```
+
+```bash
+kubectl --context $CLUSTER_1 apply -f https://github.com/camunda/c8-multi-region/blob/main/test/resources/aws/2-region/kubernetes/nginx.yml -n $CAMUNDA_NAMESPACE_1
````
I'm aware of that, currently blocked on that part.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+2. Create the Kubernetes secret within the `camunda-london` and `camunda-paris` namespace on each respective cluster.
+
+```bash
+kubectl --context $CLUSTER_1 -n $CAMUNDA_NAMESPACE_1 create secret generic elasticsearch-env-secret \
````
good catch

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+Using the same namespace on both clusters won't work since CoreDNS won't be able to distinguish between traffic targeted at the local and remote cluster.
+
+Besides the main namespaces, you will have to add `failover` namespaces for the unfortunate event of a total region loss. This is for completeness, so you don't forget to add the mapping on region recovery. The operational procedure is handled in a different [document](#). <!-- TOOD: add reference -->
+
```
good point, included since all namespaces have to be pre-created for the elastic secret.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+        reload
+        loadbalance
+    }
+    camunda-paris.svc.cluster.local:53 {
```
same as above already

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+
+The following is an example `values.yml` that forms the base and ca be further extended.
+
+Especially important within this is the `global.multiregion.regions` directive to be two for two regions. Additionally, to disable identity and optimize since those are currently not working within the setup.
```
is part of that section

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+  <summary>
+
+```yml reference title="base-values.yml"
+https://github.com/camunda/c8-multi-region/blob/aws-operational/test/resources/aws/2-region/kubernetes/camunda-values.yml
````
from the secret `extraEnvVarsSecret`?

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+The last step is to deploy Camunda 8 to both its regions, for which we chose London as region 0 and Paris as region 1. As mentioned, you may choose a different release name and should ensure you have adjusted your `base-values.yml` to consider this.
+
+```bash
+helm install camunda camunda/camunda-platform \
````
we can pin it to 9.x but should generally work with any version moving forward.

Langleu replied:
should we give version recommendations?
Starting 8.3.x, 8.4.x, 9.x?

maxdanilov replied:
maybe something to check with the Distribution team on how to refer to a Helm chart for 8.5.x stack. For now guess we can just say whatever Helm chart version will be there with 8.5.0, wdyt?

Langleu replied:
8.5.x stack --> 9x+

Langleu replied:
ah you mean that, since it's already supported why not simply try to give a version recommendation?
We can go with the current 9.2 but maybe should have a matrix from when on it's supported.
8.3.x, 8.4.x, 9.x.y

maxdanilov replied:
As we're not going to publicly advertise 8.3/8.4 support, let's use the Helm chart that refers to Camunda 8.5 y default (not sure which one will that be, but at least 9.x I guess)

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,559 @@
+  -f region1-values.yml
+```
+
+</TabItem>
````
We can. I think it would be great to have a generally applicable getting started page since we keep repeating ourselves here.

maxdanilov replied:
I was thinking about it as well. Maybe this page has EKS-specific stuff and the C8 install in the multi-region mode moved out to a separate page (and is cloud-agnostic)?

Langleu replied:
it's intertwined with AWS too much due to Elasticsearch.
What I mean is a generally applicable page on how to get started with Camunda 8, including verifying that it works.

Langleu replied:
I think it's out of scope for this PR but something in the future to have a getting started guide with C8, might even exist already, who knows.

Langleu replied:
I included a verify step with `zbctl`.
But as mentioned, I think we need a general page in the future, outside of current scope, that deals with this.

akeller commented:
I saw some discussion on this PR about documentation infrastructure, including tab functionality and plugins. Please pull in @pepopowitz for consultation since DevEx will ultimately have to maintain this, and he's the DRI for the docs infrastructure.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,592 @@
+Feel free to adjust the region and cluster names to your needs.
+
+```bash
+# The AWS regions of your cluster 1 and 2
````
shall this read cluster 0 and cluster 1?

Langleu commented:
@maxdanilov I adjusted the contens to be based on a `git clone` approach.
For the first part, removed the tab approach.
For the remainder, we could consider doing the same.

I think the only big part is the `dns chaining` solely because it's the last bit that requires manual intervention apart from just changing values.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+kubectl --context $CLUSTER_1 apply -f https://github.com/camunda/c8-multi-region/blob/main/test/resources/aws/2-region/kubernetes/nginx.yml -n $CAMUNDA_NAMESPACE_1
+```
+
+2. Try to curl each other from within the other pod:
````
```suggestion
2. Try to `curl` each other from within the other pod:
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+```
+
+</TabItem>
+<TabItem value="test-dns-chaining" label="(optional) Test DNS chaining">
````
let's drop `optional` here, I think it would be better the users check their setup always, as early in the process as possible.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+
+## Deploy Camunda 8 to the clusters
+
+<Tabs queryString="deploy">
```
Same as for previous tabs incarnation, it's not obvious that a user needs to go through steps one by one.
I suggest:
- adding `Step <N>. ` prefix to every tab name
- at the end of every tab content, add an explicit text to go to the next tab `Step <N+1>. `

Langleu replied:
I'm aware of this, I wanted to first explore whether we keep the tabs or not, now that the general procedure is shorter due to the switch to cloning the repo. As now we have a part without tabs and the rest with tabs.

Nevertheless, good suggestion with the step if we keep the tabs.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+<Tabs queryString="deploy">
+<TabItem value="elastic-secret" label="Create the secret for Elasticsearch">
+
+Elasticsearch will use S3 as a backup and restore bucket. For this, configure a Kubernetes secret to not expose those in cleartext.
```
```suggestion
Elasticsearch will need an S3 bucket for data backup and restore procedure, required during a regional failover. For this, you will need to configure a Kubernetes secret to not expose those in cleartext:
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+
+In the next part, you will configure CoreDNS to forward DNS queries for specific namespaces to the other cluster.
+
+You have to choose unique namespaces for Camunda 8 installations. The namespace for Camunda 8 installation in the cluster of region 0, needs to have a different name from the namespace for Camunda 8 installation in the cluster of region 1. This is required for proper traffic routing between the clusters.
```
```suggestion
You have to choose unique namespaces for Camunda 8 installations. The namespace for Camunda 8 installation in the cluster of region 0 (`CAMUNDA_NAMESPACE_0`), needs to have a different name from the namespace for Camunda 8 installation in the cluster of region 1 (`CAMUNDA_NAMESPACE_1`). This is required for proper traffic routing between the clusters.
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+You have to choose unique namespaces for Camunda 8 installations. The namespace for Camunda 8 installation in the cluster of region 0, needs to have a different name from the namespace for Camunda 8 installation in the cluster of region 1. This is required for proper traffic routing between the clusters.
+
+For example, you can install Camunda 8 into the `camunda-london` namespace in the `cluster-london` cluster, and `camunda-paris` namespace on the `cluster-paris` cluster.
+Using the same namespace on both clusters won't work as CoreDNS won't be able to distinguish between traffic targeted at the local and remote cluster.
```
```suggestion
Using the same namespace names on both clusters won't work as CoreDNS won't be able to distinguish between traffic targeted at the local and remote cluster.
```
I know it's a bit of tautology, but just to ensure people understand it's not about not installing Camunda twice into a same namespace, but also about not having 2 samely named namespaces in different clusters.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+
+```bash
+unset ACCESS_KEY
+unset SECRET_ACCESS_KEYv
````
```suggestion
unset SECRET_ACCESS_KEY
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+
+Now that you have created two Kubernetes clusters with Terraform, you will still have to configure various things to make the dual-region work.
+
+<Tabs queryString="cluster-setup">
```
See my comment about abother `<Tabs>` group & UX below.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+
+2. Pre-create the Camunda namespaces in each cluster. This is required to have the secrets available for Elasticsearch before installing the Helm chart.
+
+```bash
````
Does it make sense to move this whole step into a small shell script that creates the secrets?

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+</TabItem>
+<TabItem value="helm-chart" label="Camunda 8 Helm chart prerequisites">
+
+1. Within the cloned repository, navigate to `test/resources/aws/2-region/kubernetes`. This contains a dual-region example setup.
```
Reminder to rename when it happens.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+
+In our example, the `placeholder` is replaced with the following:
+
+```yaml
````
Let's make a more clear explanation that the users should not use these values, they are just for illustration.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+```bash
+http://${HELM_RELASE_NAME}-elasticsearch-master-hl.${CAMUNDA_NAMESPACE}.svc.cluster.local:9200
+# Where HELM_RELEASE_NAME is your chosen helm release name
+# and CAMUNDA_NAMESPACE is your previously defined CAMUNDA_NAMESPACE_0 and CAMUNDA_NAMESPACE_1, depending on ELASTICSEARCHREGION0 or ELASTICSEARCHREGION1
````
I'm confused about this part, what does `depending on ELASTICSEARCHREGION0 or ELASTICSEARCHREGION1` mean?

`ELASTICSEARCHREGION0` and `ELASTICSEARCHREGION1` have not been previously mentioned.

Langleu replied:
rephrased it and added an example.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+</TabItem>
+<TabItem value="deploy" label="Deploy Camunda 8">
+
+The last step is to deploy Camunda 8 to both its regions, for which we chose London as `Region 0` and Paris as `Region 1`. As mentioned, you may choose a different release name and should ensure you have adjusted your base `camunda-values.yml` to consider this:
```
```suggestion
The last step is to deploy Camunda 8 to both its regions `REGION_0` and `REGION_1`. As mentioned, you may choose a different release name and should ensure you have adjusted your base `camunda-values.yml` to consider this:
```

maxdanilov replied:
> should ensure you have adjusted your base `camunda-values.yml` to consider this:

not sure what this means, what exactly do I need to change in that yaml file?

Langleu replied:
rephrased it

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+  --version 9.3.0 \
+  --kube-context $CLUSTER_0 \
+  --namespace $CAMUNDA_NAMESPACE_0 \
+  -f base-values.yml \
```
there's no `base-values.yml` in the git repo.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+  --kube-context $CLUSTER_0 \
+  --namespace $CAMUNDA_NAMESPACE_0 \
+  -f base-values.yml \
+  -f region0-values.yml
```
there's no `region0-values.yml` in the git repo.

maxdanilov reviewed (changes requested):
Thank you, I think that it looks much cleaner and simpler with the git repo approach!
Left some comments on the points I've noticed.

maxdanilov commented:
Also, I did not leave more comments for it, but I think we should minimize the amount of times "London" or "Paris" is used in the documentation, and instead use Region 0/1 - otherwise it may be too confusing.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+
+:::
+
+#### clusters.tf
```
as mentioned in the message.
> I'm aware of branches still being wrong, blocked by missing review there + I'd move the current resources to root

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,534 @@
+
+1. Adjust any values in the `variables.tf` to your liking. For example, the target regions and their name or CIDR blocks of each cluster.
+2. Make sure that any adjustments are reflected in your [environment prerequisites](#environment-prerequisites) to ease the [in-cluster setup](#in-cluster-setup).
+3. Set up the authentication for the `AWS` provider. In the example, an AWS profile is used and provided as a Terraform variable that is defined on execution.
```
I kept it as `Set up the authentication for the `AWS` provider` since we have the exact same thing in the other guides as well, also with the same disclaimer.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
````diff
@@ -0,0 +1,534 @@
+
+Feel free to adjust the region and cluster names to your needs.
+
+```bash
````
why would you have to create a file?

I slightly adjusted it to explicitly mention the terminal.

Langleu commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,559 @@
+#### Base values.yml
+
+The following is an example `values.yml` that forms the base and ca be further extended.
+
```
let me know whether we still have to iterate on this after the updates.

Langleu commented:
@maxdanilov see the updated commits:
https://github.com/camunda/camunda-docs/pull/3430/commits/a5e3311e4fa7ca21962925ad69caa4d7dcd0a0bf
https://github.com/camunda/camunda-docs/pull/3430/commits/4bf3bb719f1fa48dde0a40926855de178c0a9eaf

- Changed to scripts which reduced the overall size of the documents.
- Removed the tabs due lower size.
- Supplied Example outputs.
- Removed London and Paris where possible. Only contained in `environment prerequisites` and any output generated from that.
     - let me know, we can adjust the output to be region / cluster 0/1
- I'm aware of links being broken due to different branch.
- I'll still have to move the resources to a different folder.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,609 @@
+- `identity.enabled: false`
+  - Identity is currently not supported.
+- `optimize.enabled: false`
+  - Optimize is currently not supported. It has a dependency on identity.
```
```suggestion
  - Optimize is currently not supported. It has a dependency on Identity.
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,609 @@
+    - The Elasticsearch endpoint for region 0.
+  - `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCHREGION1_ARGS_URL`
+    - The Elasticsearch endpoint for region 1.
+- Following cluster of 8 is recommend for a dual-region setup
```
```suggestion
- A cluster of 8 Zeebe brokers (4 in each of the regions) is recommended for the dual-region setup
```

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,609 @@
+  - `ZEEBE_BROKER_EXPORTERS_ELASTICSEARCHREGION1_ARGS_URL`
+    - The Elasticsearch endpoint for region 1.
+- Following cluster of 8 is recommend for a dual-region setup
+  - `zeebe.clusterSize: 8`
```
did you double check with Consulting/Zeebe that 8/8/8 is the correct setup? I had a vague memory of the hack week that not all of these values are 8, but I could be wrong.

Langleu replied:
you're right, sorry.
it's 8/8/4, my `zbctl status` shows the correct one.

maxdanilov commented on docs/self-managed/platform-deployment/helm-kubernetes/platforms/amazon-eks/dual-region.md:
```diff
@@ -0,0 +1,609 @@
+
+#### vpc-peering.tf
+
+For a multi-region setup, you need to have the [virtual private cloud (VPC)](https://aws.amazon.com/vpc/) peered to route traffic between them using private IPv4 addresses and not publicly route the traffic and expose it. For more information, review the [AWS documentation on VPC peering](https://docs.aws.amazon.com/vpc/latest/peering/what-is-vpc-peering.html).
```
```suggestion
For a multi-region setup, you need to have the [virtual private cloud (VPC)](https://aws.amazon.com/vpc/) peered to route traffic between regions using private IPv4 addresses and not publicly route the traffic and expose it. For more information, review the [AWS documentation on VPC peering](https://docs.aws.amazon.com/vpc/latest/peering/what-is-vpc-peering.html).
```

maxdanilov reviewed (approved):
Looks good overall, thank you. Nice to see the guide getting leaner with the scripts.

Just leaving small comments.
